content/docs/arvo/dill/tasks.md

```diff
@@ -0,0 +1,343 @@
+
+#### Returns
+
+Dill does not return a `gift` in response to a `%flog` `task`.
+
+## `%flow`
+
+```hoon
+[%flow p=@tas q=(list gill:gall)]
+```
+
+Terminal config.
+
+This `task` is not used.
+
+## `%hail`
+
+```hoon
+[%hail ~]
+```
+
+Refresh.
+
+Dill converts a `%hail` `task` into a `%hey` [$dill-belt](/docs/arvo/dill/data-types#dill-belt) and `%poke`s the session handler (typically `drum`) with it to handle the refresh.
+
+This `task` would not be used from userspace.
+
+#### Returns
+
+Dill returns no `gift` in response to a `%hail` `task`.
+
+## `%heft`
+
+```hoon
+[%heft ~]
+```
+
+Produce memory report.
+
+When Dill receives a `%heft` `task` it passes Arvo a `%whey` `waif` to obtain a memory report and prints it to the terminal.
+
+#### Returns
+
+Dill does not return a `gift` in response to a `%heft` `task`.
+
+## `%hook`
+
+```hoon
+[%hook ~]
+```
+
+This terminal hung up.
+
+This task is not used.
+
+## `%harm`
+
+```hoon
+[%harm ~]
+```
+
+All terminals hung up.
+
+This `task` is not used.
+
+## `%init`
+
+```hoon
+[%init ~]
+```
+
+This `task` is called only once, when Arvo first enters the [adult stage](/docs/arvo/overview#structural-interface-core). It performs initial setup for Dill, such as setting the width of the console.
+
+Note that this is not actually the first `task` passed to Dill - see [%boot](#%boot).
+
+This `task` would not be used from userspace.
+
+#### Returns
+
+Dill does not return a `gift` in response to a `%init` `task`.
+
+## `%meld`
+
+```
+[%meld ~]
+```
+
+Deduplicate persistent state.
+
+Dill asks the runtime to perform the memory deduplication.
+
+#### Returns
+
+Dill does not return a `gift` in response to a `%meld` `task`.
+
+## `%noop`
+
+```hoon
+[%noop ~]
+```
+
+No operation.
+
+A `%noop` `task` does nothing, as the name implies.
+
+#### Returns
+
+Dill does not return a `gift` in response to a `%noop` `task`.
+
+## `%pack`
+
+```hoon
+[%pack ~]
+```
+
+Defragment persistent state.
+
+Dill asks the runtime to perform the defragmentation.
+
+#### Returns
+
+Dill does not return a `gift` in response to a `%meld` `task`.
+
+## `%talk`
+
+```hoon
+[%talk p=tank]
+```
+
+This `task` is not used.
+
+## `%text`
+
+```hoon
+[%text p=tape]
+```
+
+Print `tape` to terminal.
+
+Upon receiving a `%text` `task`, Dill will convert the `tape` given in `p` to a `(list @c)` and give it to the runtime in a `%blit` `gift` including a `%lin` [$blit](/docs/arvo/dill/data-types#blit).
+
+#### Returns
+
+Dill does not return a `gift` in response to a `%text` `task`, but it does give a `%blit` `gift` to the runtime which looks like:
+
+```hoon
+[%blit p=(list blit)]
+```
+
+## `%view`
+
+```hoon
+[%view session=~]
+```
+
+Watch session.
+
+A `%view` `task` subscribes for a copy of all `%blit`s which Dill sends to the Unix terminal. This `task` is used by the `%herm` app so it can convert the `$blit`s to JSON and mirror the Unix terminal in the web terminal.
```

Comment:
> subscribes for a copy of ...

Hmm, perhaps this is pedantic, but I wouldn't describe it this way. Rather, a `%view` by the runtime is sort of implicit in `%boot`. Sending a `%view` merely adds another subscriber to the same endpoint that the runtime/Unix terminal happens to be listening on. Herm then, too, is just subscribing to dill sessions, rather than the specific thing that the runtime is interested in. (Mirroring is presently indeed the result though.)